Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -199,6 +199,8 @@ int GameListModel::rowCount(const QModelIndex& parent) const
 
 int GameListModel::columnCount(const QModelIndex& parent) const
 {
+  if (parent.isValid())
+    return 0;
```

Comment:
This looks really weird to me (as if it should always return zero from now on). _When_ is `parent` ever _not_ valid? Why is it being passed if it could ever be _not_ valid?

Reply:
No idea honestly. I based it solely on the documentation and a similar code in `rowCount`:

https://doc.qt.io/qt-5/qabstractitemmodel.html#columnCount

`Note: When implementing a table based model, columnCount() should return 0 when the parent is valid.`

Our model is table based, so I decided to follow the docs.